src/app/pages/OpenWalletPage/Features/FromLedger/index.tsx

```diff
@@ -1,56 +1,100 @@
-import { importAccountsActions } from 'app/state/importaccounts'
 import { Box } from 'grommet/es6/components/Box'
+import React, { useEffect } from 'react'
+import { Header } from 'app/components/Header'
+import { ButtonLink } from '../../../../components/ButtonLink'
 import { Button } from 'grommet/es6/components/Button'
-import { Heading } from 'grommet/es6/components/Heading'
-import React from 'react'
+import { Text } from 'grommet/es6/components/Text'
+import { canAccessBle, canAccessNavigatorUsb } from '../../../../lib/ledger'
 import { useTranslation } from 'react-i18next'
-import { useDispatch, useSelector } from 'react-redux'
-import { ImportAccountsSelectionModal } from 'app/pages/OpenWalletPage/Features/ImportAccountsSelectionModal'
-import { selectShowAccountsSelectionModal } from 'app/state/importaccounts/selectors'
-import { Header } from 'app/components/Header'
-import { WalletType } from 'app/state/wallet/types'
+import { useNavigate } from 'react-router-dom'
+
+type SelectOpenMethodProps = {
+  webExtensionLedgerAccess?: () => void
+}
 
-export function FromLedger() {
+export function FromLedger({ webExtensionLedgerAccess }: SelectOpenMethodProps) {
   const { t } = useTranslation()
-  const dispatch = useDispatch()
-  const showAccountsSelectionModal = useSelector(selectShowAccountsSelectionModal)
+  const [supportsUsbLedger, setSupportsUsbLedger] = React.useState<boolean | undefined>(true)
+  const [supportsBleLedger, setSupportsBleLedger] = React.useState<boolean | undefined>(true)
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    async function getLedgerSupport() {
+      const usbLedgerSupported = await canAccessNavigatorUsb()
+      const bleLedgerSupported = await canAccessBle()
+
+      if (usbLedgerSupported && !bleLedgerSupported && !webExtensionLedgerAccess) {
+        navigate('usb')
+      } else if (bleLedgerSupported && !usbLedgerSupported && !webExtensionLedgerAccess) {
+        navigate('ble')
+      }
```

Comment:
I would rather see both choices listed and the disabled one have an explanation. I think it would be confusing to see different behavior on different device.

Reply:
Right, make more sense, than the current redirects.